R/gating-methods.R

```diff
@@ -381,8 +381,8 @@ roxygen_parameter <- function() {
       }
     }
 
-    # For gate_quad methods, need to filter down to just the gates that were asked for
-    if(names(x) %in% c("quadGate.seq", "gate_quad_sequential", "quadGate.tmix", "gate_quad_tmix")){
+    # For gate_quad methods, need to filter down to just the gates that were asked for if pop != "*"
+    if(!all(popName %in% "*") & names(x) %in% c("quadGate.seq", "gate_quad_sequential", "quadGate.tmix", "gate_quad_tmix")){
```

Comment:
```suggestion
    if(!all(popName %in% "*") && any(names(x) %in% c("quadGate.seq", "gate_quad_sequential", "quadGate.tmix", "gate_quad_tmix"))){
```

Reply:
Not sure exactly what the cardinality of `names(x)` is. Somehow this if-block needs to ensure it returns a scalar TRUE or FALSE

Reply:
gtMethod object is parsed from the gatingTemplate CSV file without any string processing so its names() slot will always have length equal to one. Also, `gs_add_gating_method()` actually converts the inputs into to a CSV before gating so it will be treated the same way. I can't think of an instance where a user would be able to successfully pass in multiple gating methods for a single gating step, so I think we are OK without adding additional security here.

Reply:
Ok, we should still use the short-circuiting `&&` operator here though
```suggestion
    if(!all(popName %in% "*") && names(x) %in% c("quadGate.seq", "gate_quad_sequential", "quadGate.tmix", "gate_quad_tmix")){
```